lib/truffle/truffle/patching.rb

```diff
@@ -1,34 +1,116 @@
 module Truffle::Patching
-  TRUFFLE_PATCHES_DIRECTORY = "#{Truffle::Boot.ruby_home}/lib/patches"
-  TRUFFLE_PATCHES           = Dir.glob("#{TRUFFLE_PATCHES_DIRECTORY}/**/*.rb").
-      select { |path| File.file? path }.
-      map { |path| [path[(TRUFFLE_PATCHES_DIRECTORY.size + 1)..-4], true] }.
-      to_h
+  DIR     = "#{Truffle::Boot.ruby_home}/lib/patches"
+  # Allowed operations are: :before, :after, :instead, :ignore
+  PATCHES = { "bundler"                                 => :before,
+              "bundler/version"                         => :after,
+              "bundler/cli/exec"                        => :after,
+              "bundler/compact_index_client/updater"    => :after,
+              "bundler/current_ruby"                    => :after,
+              "bundler/dependency"                      => :after,
+              "bundler/fetcher/downloader"              => :after,
+              "bundler/fetcher/compact_index"           => :after,
+              "bundler/ruby_version"                    => :after,
+              "bundler/source/rubygems"                 => :after,
+
+              "rubygems"                                => :before,
+              "rubygems/ext"                            => :after,
+              "rubygems/package"                        => :after,
+              "rubygems/remote_fetcher"                 => :after,
+              "rubygems/request"                        => :after,
+              "rubygems/request_set/gem_dependency_api" => :after,
+
+              "rspec/support"                           => :after,
+
+              "unf/normalizer_cruby"                    => :before,
+  }
+
+  PATCHES.each do |file, _|
+    unless File.exist? File.join(DIR, file) + '.rb'
+      raise "file #{file} is missing in #{DIR}"
+    end
+  end
+
+  Dir["#{DIR}/**/*.rb"].each do |file|
+    relative_path = file[(DIR.size + 1)..-4]
```

Comment:
It's not just a basename though, the result is e.g. `bundler/version` from `/Users/pitr/Workspace/labs/truffleruby-mx/lib/ruby/gems/2.3.0/gems/bundler-1.14.6/lib/bundler/version.rb` I could wrap it in a method though.